### DISCUSSION TITLE
Provide API that produces information as to which platforms would be scanned if the scan command was run

### DISCUSSION THREAD

**Discussion by binarymist** [Checkmarx/kics#4912] [Ideas] [answered]
HI.

Are there any plans to provide API that produces information as to which platforms would be scanned if the scan command was run.
We are looking to not initiate a scan at all if specific or all supported platforms are not represented in the source under test.

Thanks.

**Comment by cx-nuno-oliveira** [comment, marked as answer]
Hi @binarymist 
Whenever a scan is triggered, we output the technologies that will be scanned (the ones we could identify).

We have the `-t, --type strings` flag that will scan only specific technologies. This forces the scan of specified techs even if the relevant files are not present (which is the opposite of your request).

We are proposing to change its behavior to not force the scan if the relevant files are not there. But still seems not what you want.

Please, open a feature request for this, and provide us some use cases so we can prioritize it, accordingly.
Or, consider to contribute with such an API! We would be glad to help you in the process!

**Reply by binarymist** [reply]
Hi @nunoocx.

Actually we check for the presense of the file types that kics supports prior to a scan in another service that I wasn't aware of, so that should be all we need currently.

Thanks.

**Reply by cx-nuno-oliveira** [reply]
Great to know that!
If something else is needed, don't hesitate in reaching us!
Thanks

**Comment by niro-lightspin** [comment]
Hi, we're also looking for such API so we can clear all the non supported extensions from the repo before scanning it (as we don't like to show files with 0 findings)
Was there a feature request eventually?
Is such API available today?
Is there some place in the code where such list is available?

**Reply by gabriel-cx** [reply]
Hi @niro-lightspin ,

From the [latest version ](https://github.com/Checkmarx/kics/releases/tag/v1.6.11) release we started to provide as a .zip file information about the technologies we support.
If you go to the previous link and scroll down, in the **_assets_** section you will see a .zip file named _**extracted-info**_.
Inside that .zip you can find information about the technologies we support, such as: _**Platforms**_ (Terraform, Dockerfile, etc.) and _**Cloud Providers**_ (AWS, Azure, etc.).

Is this the information you need?

**Reply by niro-lightspin** [reply]
It is but i'm missing the file extensions, for example:
{
    "platforms": [
        {
            "id": 1,
            "platform": "ServerlessFW",
            "extension: ".yml"
        },
...

**Reply by niro-lightspin** [reply]
Or maybe this:
{
"platforms": [
{
"id": 1,
"platform": "ServerlessFW",
"extensions: [".yml"]
},
...

**Reply by gabriel-cx** [reply]
The supported file extensions can be checked in the code itself, [here, line 70](https://github.com/Checkmarx/kics/blob/1d1628ff80c8a1fae8430d13aff36b7ec827bd54/pkg/analyzer/analyzer.go#L70).

I think this is the info you are looking for!

**Reply by niro-lightspin** [reply]
Yes, thanks @gabriel-cx

**Reply by gabriel-cx** [reply]
No problem! Feel free to contact us if you need more information!
